File: app/src/main/java/com/axaet/butterknife/activity/MainActivity.java
package com.axaet.butterknife.activity;

import android.app.Application;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.alibaba.android.arouter.core.LogisticsCenter;
import com.alibaba.android.arouter.facade.Postcard;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;
import com.axaet.butterknife.BuildConfig;
import com.axaet.butterknife.R;
import com.axaet.butterknife.application.MyApplication;
import com.axaet.module.base.common.ARouterURL;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * @author android
 */
@Route(path = ARouterURL.MAIN_ACTIVITY)
public class MainActivity extends AppCompatActivity {

    /**
     * 不同变体不同数值
     */
    public static final boolean THREE_PARTY_LOGIN = BuildConfig.THREE_PARTY_LOGIN;

    public static final int REQUEST_CODE = 0x01;
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        Application application = MyApplication.getInstance().getApplication();
        Log.i(TAG, THREE_PARTY_LOGIN + "   app name:  " + application.getClass().getName());
    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_CODE && resultCode == RESULT_OK && data != null) {
            String result = data.getStringExtra("result");
            Toast.makeText(this, result, Toast.LENGTH_SHORT).show();
        }
    }

    @OnClick({R.id.btnShow, R.id.button})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btnShow:
                try {
                    Postcard postcard = ARouter.getInstance().build(ARouterURL.LOGIN_ACTIVITY);
                    LogisticsCenter.completion(postcard);
                    Class<?> destination = postcard.getDestination();
                    Log.i(TAG, destination.getName());
                    startActivityForResult(new Intent(this, destination), REQUEST_CODE);
                } catch (Exception e) {
                    Toast.makeText(this, "没有依赖该module", Toast.LENGTH_SHORT).show();
                }
                break;
            case R.id.button:
                ARouter.getInstance().build(ARouterURL.NEWS_MAIN)
                        .withString("msg", "从主module传递参数给其他module")
                        .withBoolean("boy", true)
                        .navigation();
                break;
            default:
        }
    }
}

File: module_news/src/main/java/com/axaet/news/MainActivity.kt
package com.axaet.news

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.launcher.ARouter
import com.axaet.module.base.common.ARouterURL
import kotlinx.android.synthetic.main.news_activity_main.*

@Route(path = ARouterURL.NEWS_MAIN)
class MainActivity : AppCompatActivity() {

    @JvmField
    @Autowired
    var boy: Boolean = false

    @JvmField
    @Autowired
    var msg: String? = null


    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.news_activity_main)
        //获取其他组件传递过来的数组需要注入
        ARouter.getInstance().inject(this)
        Log.i(TAG, "boy=$boy")
        mTvShow.text = "$msg   boy=$boy"
    }

    companion object {
        const val TAG = "MainActivity"
    }
}
